src/Core/Validation/FieldValidation/BigIntFieldValidator.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace SilverStripe\Core\Validation\FieldValidation;
+
+use SilverStripe\Core\Validation\FieldValidation\IntFieldValidator;
+
+class BigIntFieldValidator extends IntFieldValidator
+{
+    /**
+     * The minimum value for a signed 64-bit integer.
+     * Defined as string instead of int otherwise will end up as a float
+     * on 64-bit systems if defined as an int
```

Comment:
```suggestion
     * Defined as string instead of int otherwise will end up as a float
     * on 64-bit systems
```
No need to mention that bit twice.
Also, just to confirm, did you mean "end up as a float on 32-bit systems"? Since this is a 64-bit int I'd expect it to be fine for 64-bit system, but fail on 32-bit systems?

Reply:
Removed

Comment is correct, it's on 64 bit systems

```php
class C {
  private const MIN_64_BIT_INT = -9223372036854775808;
  public function f() { return self::MIN_64_BIT_INT; }
}
// double(-9.2233720368548E+18)
```